packages/cli/src/types.ts

```diff
@@ -0,0 +1,126 @@
+// Reference to and optional definition of a type alias:
+// export type ${name} = ${definition}
+import { keyof } from 'io-ts';
+
+export class TypeAlias {
+  constructor(
+    public readonly name: string,
+    public readonly definition?: string,
+  ) {}
+
+  defined(): boolean {
+    return this.definition !== undefined;
+  }
+
+  declaration(): string {
+    if (!this.defined()) {
+      return '';
+    }
+    return `export type ${this.name} = ${this.definition}\n`;
+  }
+}
+
+// Default types
+const String = new TypeAlias('string');
+const Number = new TypeAlias('number');
+const Boolean = new TypeAlias('boolean');
+const Date = new TypeAlias('Date');
+const Bytes = new TypeAlias('Buffer');
+const Json = new TypeAlias(
+  'Json',
+  'null | boolean | number | string | Json[] | { [key: string]: Json }',
+);
+
+export const DefaultTypeMapping = Object.freeze({
+  // Integer types
+  int2: Number,
+  int4: Number,
+  int8: Number,
+  smallint: Number,
+  int: Number,
+  bigint: Number,
+
+  // Precision types
+  real: Number,
+  float4: Number,
+  float: Number,
+  float8: Number,
+  numeric: Number,
+  decimal: Number,
+
+  // Serial types
+  smallserial: Number,
+  serial: Number,
+  bigserial: Number,
+
+  // Common string types
+  uuid: String,
+  text: String,
+  varchar: String,
+  char: String,
+
+  // Bool types
+  bit: Boolean, // TODO: better bit array support
+  bool: Boolean,
+  boolean: Boolean,
+
+  // Extra types
+  date: Date,
+  money: Number,
+
+  // JSON types
+  json: Json,
+  jsonb: Json,
+
+  // Bytes
+  bytea: Bytes,
+});
+
+export type PostgresType = keyof typeof DefaultTypeMapping;
```

Comment:
Lets keep the naming DB neutral to make porting this to other DBs a little easier.